stl/inc/xstring

```diff
@@ -2843,9 +2861,11 @@ private:
 #endif // _HAS_CXX20
                     _Traits::copy(_Unfancy(_New_ptr), _Old_ptr, _My_data._Mysize);
                     if (_My_data._Myres >= _BUF_SIZE) { // Need to deallocate old storage
+                        _ASAN_STRING_REMOVE(*this);
                         _Al.deallocate(_My_data._Bx._Ptr, _My_data._Myres + 1);
                         _My_data._Bx._Ptr = _New_ptr;
                     } else {
+                        _ASAN_STRING_SWITCH_TO_LARGE(_My_data);

```

Comment:
Something is fishy here.

At the end of the function we call `_ASAN_STRING_CREATE(*this);` Why do we already have a valid ASAN region here and still do that?

Reply:
`_ASAN_STRING_SWITCH_TO_LARGE` creates a valid ASAN region _inside the SSO buffer_ for the pointer. `_ASAN_STRING_CREATE` then creates a valid ASAN region in the large buffer.